File: flutter/lib/features/outbound/presentation/pages/long_term/information/selection_day.dart
// 🐦 Flutter imports:
import 'package:flutter/material.dart';
import 'package:flutter/services.dart';

// 🌎 Project imports:
import 'package:rotary_nl_rye/core/prop.dart';
import 'package:rotary_nl_rye/features/uniform_widgets/back_button.dart';

class SelectionDayPage extends StatefulWidget {
  @override
  _SelectionDayPageState createState() => _SelectionDayPageState();
}

class _SelectionDayPageState extends State<SelectionDayPage> {
  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        systemOverlayStyle:
            MediaQuery.of(context).platformBrightness == Brightness.light
                ? SystemUiOverlayStyle.dark
                : SystemUiOverlayStyle.light,
        backgroundColor: Colors.transparent,
        elevation: 0.0,
        leading: UniformBackButton(),
        title: Text(
          'Selectie dag',
          textScaler: TextScaler.linear(1),
          style: TextStyle(color: Palette.indigo, fontWeight: FontWeight.bold),
        ),
      ),
      body: ListView(
        padding: EdgeInsets.symmetric(horizontal: 20, vertical: 15),
        children: <Widget>[
          buildSectionTitle('Wat moet ik doen voor de selectie dag:'),
          buildContentText('- Ik zou gewoon je best en doen en jezelf zijn.'),
          buildSectionTitle('Dit is wat je deze dag kan verwachten:'),
          buildContentText(
              '- Je krijgt een interview, een groepsgesprek, een discussie en een test over je kennis van Nederland.'),
          buildContentText(
              '- Dit zijn een paar voorbeeld vragen uit het interview tijdens de selectie dag. We gaan je niet alles vertellen, maar zo krijg je een beetje een idee.'),
          buildSectionTitle('Voorbeeld vragen:'),
          buildContentText(
              '- Wat betekent volgens jou het zijn van Ambassadeur voor Rotary'),
          buildContentText('- Wie is je rolmodel, voor wie heb je bewondering'),
          buildContentText('- Wat was de gelukkigste/mooiste dag in je leven'),
          buildContentText('- Op welke eigenschap ben je het meest trots'),
          buildContentText(
              '- Wat denk je dat het moeilijkste is als je een jaar in het buitenland bent'),
          buildFooter(),
        ],
      ),
    );
  }

  Widget buildSectionTitle(String text) {
    return Padding(
      padding: const EdgeInsets.only(top: 25.0),
      child: Text(
        text,
        style: TextStyle(fontSize: 14.0, fontWeight: FontWeight.bold),
      ),
    );
  }

  Widget buildContentText(String text) {
    return Padding(
      padding: const EdgeInsets.only(top: 10.0),
      child: Text(
        text,
        style: TextStyle(fontSize: 14.0),
      ),
    );
  }

  Widget buildFooter() {
    return Column(
      children: [
        Padding(
          padding: const EdgeInsets.only(top: 30.0),
          child: Center(
            child: Image.asset(
              'assets/image/rotary_blue.png',
              height: 55.0,
            ),
          ),
        ),
        Padding(
          padding: const EdgeInsets.only(top: 15.0),
          child: Center(
            child: Text(
              'Update: 31 May 2021',
              style: TextStyle(color: Color(0xFF777777)),
            ),
          ),
        ),
        SizedBox(height: 60),
      ],
    );
  }
}

File: rotary-yep-nl/features/student-info/content/outbound/long-term/selection-day.ts
import type { InfoPageContent } from "../../../types";

export const selectionDayContent: InfoPageContent = {
  pageKey: "outbound-selection-day",
  header: {
    icon: "people-outline",
    title: "Selectie dag",
    subtitle: "Wat je kunt verwachten tijdens de selectiedag voor Rotary Youth Exchange",
  },
  sections: [
    {
      id: "what-to-do",
      icon: "checkmark-circle-outline",
      title: "Wat moet ik doen voor de selectie dag:",
      blocks: [
        {
          id: "advice-card",
          type: "card",
          content: "Ik zou gewoon je best en doen en jezelf zijn.",
        },
      ],
    },
    {
      id: "what-to-expect",
      icon: "eye-outline",
      title: "Dit is wat je deze dag kan verwachten:",
      blocks: [
        {
          id: "activities-card",
          type: "card",
          content:
            "Je krijgt een interview, een groepsgesprek, een discussie en een test over je kennis van Nederland.",
        },
        {
          id: "preview-card",
          type: "card",
          content:
            "Dit zijn een paar voorbeeld vragen uit het interview tijdens de selectie dag. We gaan je niet alles vertellen, maar zo krijg je een beetje een idee.",
        },
      ],
    },
    {
      id: "example-questions",
      icon: "help-circle-outline",
      title: "Voorbeeld vragen:",
      blocks: [
        {
          id: "question-1",
          type: "card",
          accentColor: "primary",
          content: "Wat betekent volgens jou het zijn van Ambassadeur voor Rotary",
        },
        {
          id: "question-2",
          type: "card",
          accentColor: "primary",
          content: "Wie is je rolmodel, voor wie heb je bewondering",
        },
        {
          id: "question-3",
          type: "card",
          accentColor: "primary",
          content: "Wat was de gelukkigste/mooiste dag in je leven",
        },
        {
          id: "question-4",
          type: "card",
          accentColor: "primary",
          content: "Op welke eigenschap ben je het meest trots",
        },
        {
          id: "question-5",
          type: "card",
          accentColor: "primary",
          content: "Wat denk je dat het moeilijkste is als je een jaar in het buitenland bent",
        },
      ],
    },
  ],
};
